### ISSUE TITLE
OpenMS simulation and ImageCreator

### ISSUE THREAD

**Issue by Fjnejad** [OpenMS/OpenMS#3481]
Dear OpenMS People,

I am using OpenMS to simulate and analyze LC_MS data.
I use the following code to generate LC_MS data:
MSSimulator -in PATH_TO_FASTA_FILE/MS_$i.fasta -out /PTH_TO_OUTPUT_MZML_FILE/MS_$i.mzML -threads 12
but I get the following messages for some of the samples:
1." RT prediction gave 'invalid' results for 2904 peptide(s), making them unobservable."
2."FATAL: uncaught exception!

last entry in the exception handler: 
exception of type InvalidValue occured in line 121, function static FactoryProduct* OpenMS::Factory<FactoryProduct>::create(const OpenMS::String&) [with FactoryProduct = OpenMS::BaseModel<1u>] of /home/USER/Applications/OpenMS/src/openms/include/OpenMS/CONCEPT/Factory.h
error message: the value 'GaussModel' was used but is not valid; This FactoryProduct is not registered!

/PATH_TO_MY_SCRIPT/generateMS.sh: line 8: 16457 Aborted                 (core dumped) /home/USER/Applications/OpenMS-build/bin/MSSimulator -in /PATH_TO_FASTA_FILE/MS_9.fasta -out /PATH_TO_OUT_PUT_MZML_FILE/MS_9.mzML -threads 12"
Also, in some cases it just stops simulating and I have to kill the process and restart my script.
I have also attaches a sample fasta file.

After simulating the LC_MS data and preprocessing them I convert them to images and then do some analysis on them. Now I have to get the mzValue and the intensity of the peaks shown in the images for the rest of my analysis. How could I get that information? (as an example I have got 501*50 matrices after peak picking which are converted to 512*512 png images).

Many thanks in advance,

[MS_2810.txt](https://github.com/OpenMS/OpenMS/files/2027538/MS_2810.txt)

**Comment by jpfeuffer** [comment]
1) The predicted retention time for them lies outside of the min and max of your gradient time or is smaller than 0. This can happen with the SVM used for prediction. Play around with the gradient time parameter and/or train your own model if possible. This should be just a warning and most of the time you can ignore it. But 2904 peptides could be a bit much.

2) This could be a multithreading issue. Does it go away if you remove the threads parameter (maybe on a subset). We also had some recent fixes for multithreading. You could try a develop version if your don't do so already.

3) Depends on what you see in those images. Is it the LC-MS map (or a crop of it) or some fragment spectra?

**Comment by hroest** [comment]
> exception of type InvalidValue occured in line 121, function static FactoryProduct* OpenMS::Factory::create(const OpenMS::String&) [with FactoryProduct = OpenMS::BaseModel<1u>] of /home/USER/Applications/OpenMS/src/openms/include/OpenMS/CONCEPT/Factory.h
> error message: the value 'GaussModel' was used but is not valid; This FactoryProduct is not registered!

sounds like the same issue as #3431 and I think this issue will be addressed #3439 see https://github.com/OpenMS/OpenMS/pull/3439/files#diff-93ac00a0388dbbcdacf2da5243d8dbacR120

**Comment by Fjnejad** [comment]
@jpfeuffer thank you very much for your reply. the multi-thread issue. is solved now.
about the images:
I simulate the LC_MS spectra using a fasta file including 50 random proteins. and fetch the peaks ( for each sample a matrix of 501x50, 50 peaks in each row). Then the output matrix is converted to a 512x512 image.
 Thanks

**Comment by Fjnejad** [comment]
I need to have two different classes of data with different peaks, so I chose two groups of different proteins (each 50) for each class.
yes, each sample is a replication and that 501 is the retention time.
TO make it may be more clear, I have attached a fasta file, the peaks mzML file and the images from the original spectra and the image from the peaks.

[SampleMS.zip](https://github.com/OpenMS/OpenMS/files/2047708/SampleMS.zip)
